include/Utility/id_t.hpp

```diff
@@ -0,0 +1,9 @@
+#pragma once
+
+namespace CXXGraph {
+#ifdef CXXGRAPH_ID_TYPE
+typedef CXXGRAPH_ID_TYPE id_t;
+#else
+typedef size_t id_t;
+#endif
+}
```

Comment:
Default to `size_t` as a compromise between performance (as `size_t` is a platform-dependent size), and hash collision rate.

Users can define `CXXGRAPH_ID_TYPE` if they want a different type.

Reply:
It's ok for generality, but the id should be ever an INTEGER value, or we have problem in the code.
You think is possible to check this when redefined?

Reply:
I'm not sure if it's possible to have the preprocessor ensure unsigned-ness. I'll look into it